### PR TITLE
RFC: Lint CSV files on GitHub Actions

### PR DESCRIPTION
**Pull request by jonafato**
Port validation from TravisCI to GitHub Actions and use `vladiate` for field-level validation (specific details in commit messages).

### CODE REVIEW

**Comment by jonafato** [comment]
@davidfischer @eumiro Mentioning you both here, since the three of us are the ones that have modified the `lint-csv.py` file according to GitHub history. My main goals here were to get the linting step up and running again and to add more granular validation where applicable (if `csvclean` supports field-level validation, or if there's another approach we should use, I'm open to suggestions).

**Comment by davidfischer** [comment]
I wasn't extremely familiar with CSV validation. I helped work on this validation because the key thing was to make sure the CSV is actually a valid CSV and therefore can be displayed as one nicely in GitHub. Lots of folks just try to add a row and what they add may or may not be a valid CSV. That's what I tried to prevent.

File: 2023.csv [hunk 1]
```diff
@@ -17,7 +17,7 @@ PyCon Taiwan,2023-09-02,2023-09-03,"Taipei, Taiwan",TWN,,,,https://tw.pycon.org/
 Kiwi PyCon XII,2023-09-15,2023-09-17,"Waihōpai (Invercargill), Aotearoa, New Zealand",NZL,Ascot Park Hotel,,2023-05-19,https://kiwipycon.nz/,https://kiwipycon.nz/call-for-proposals,https://kiwipycon.nz/sponsorship
 PyCon Uganda,2023-09-21,2023-09-23,"MoTIV, Kampala, Uganda",UGA,,,,https://ug.pycon.org,https://ug.pycon.org/speakers,https://ug.pycon.org/sponsors
 PyCon UK,2023-09-22,2023-09-25,"Cardiff, United Kingdom",GBR,Cardiff City Hall,,,https://2023.pyconuk.org/,,https://2023.pyconuk.org/sponsorship/
-PyGotham TV 2023,2023-10-06,2023-10-07,"New York, New York, United States of America",USA,Online,,2023-05-15,https://2023.pygotham.tv,https://cfp.pygotham.tv,
```
Comment (davidfischer) [inline 2023.csv:20]: Was this accidental?

**Review by davidfischer — APPROVED** [review]
I reviewed the code and it looks good. I will be honest that I'd never heard of the vladiate module (looks cool though) and I didn't run it myself although I did verify the "checks" result in GitHub actions.

File: 2023.csv (hunk 1 above)
Comment (jonafato) [reply 2023.csv:20]: Sure was! (I'm an organizer for PyGotham and was using that row as a test, oops.)

Comment (jonafato) [reply 2023.csv]: Thanks for catching that. Fixed.

**Review by darjeeling — COMMENTED** [review]
how about using this module? https://pypi.org/project/iso3166/
I think we can reduce changes and depend what they made

**Comment by jonafato** [comment]
> how about using this module? https://pypi.org/project/iso3166/ I think we can reduce changes and depend what they made

Great suggestion, I've just updated the code to use that library instead of copying the list from Wikipedia.

**Review by darjeeling — DISMISSED** [review]

**Comment by darjeeling** [comment]
@jonafato another suggestion.

I think linting is good things to be done better first.

after that how about add comments to fix on PR? 
if PR has changes on csv. with linter can can give them suggestion to fix.

**Comment by jonafato** [comment]
@darjeeling This linter doesn't do any kind of auto-fixing, but if there are reliable ways to apply or suggest fixes, I think that would be a nice followup pull request.

**Comment by darjeeling** [comment]
I tested with this and looks fine.
```
(conferences) ➜  conferences git:(lint-csv-on-gha) ✗ python linting/lint-csv.py

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2017.csv'))
Passed! :)

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2018.csv'))
Passed! :)

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2019.csv'))
Passed! :)

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2020.csv'))
Passed! :)

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2021.csv'))
Passed! :)

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2022.csv'))
Passed! :)

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2023.csv'))
Passed! :)

Validating ConferencesValidator(source=LocalFile('/Users/dj/github/darjeeling/conferences/2024.csv'))
Passed! :)
```

**Review by darjeeling — APPROVED** [review]
I think we can go merge.

**Pull request merged**